provider/pkg/host/host.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"golang.org/x/net/context"
+)
+
+// HostClient is the interface that the provider uses to communicate with the Pulumi engine.
+//
+//nolint:golint
```

Comment:
Curious as to why we need nolint here?

Reply:
The linter wants to rename `HostClient` to `Client` (stutter) but I would prefer that the interface name resemble the concrete `HostClient`.